CMake/sitkMaxDimensionOption.cmake

```diff
@@ -0,0 +1,29 @@
+#
+# A common CMake file for consistently initializing and verifying the
+# SimpleITK_MAX_DIMENSION CMake variable.
+#
+
+
+set(SimpleITK_MAX_DIMENSION_DEFAULT 4)
+# 1900 = VS 14.0 (Visual Studio 2015)
+if(MSVC AND MSVC_VERSION VERSION_LESS 1900)
+  set( SimpleITK_MAX_DIMENSION_DEFAULT 3 )
+endif()
+
+if (DEFINED SimpleITK_4D_Image)
+  message(WARNING "SimpleITK variable \"SimpleITK_4D_Image\" is \
+                   deprecated set \"SimpleITK_MAX_DIMENSION\" instead." )
+  if (NOT SimpleITK_4D_Image)
+    set( SimpleITK_MAX_DIMENSION_DEFAULT 3 )
+  endif()
+endif()
+
+set( "SimpleITK_MAX_DIMENSION" "${SimpleITK_MAX_DIMENSION_DEFAULT}"
+  CACHE STRING "Add Image and I/O support up to max dimension (integer)." )
+mark_as_advanced( SimpleITK_MAX_DIMENSION )
+
+if (NOT SimpleITK_MAX_DIMENSION MATCHES "^[0-9]+$")
```

Comment:
I think the regular expression should be "^[1-9][0-9]+$" number shouldn't start with a zero.

Reply:
Accepting leading zeros for numbers is standard numerical input behavior. The check for a minimum value is done below. I don't see any potential problem.